src/Bicep.LangServer/Handlers/BicepDecompileForPasteCommandHandler.cs

```diff
@@ -128,34 +152,19 @@ private async Task<ResultAndTelemetry> TryDecompileForPaste(string json, bool qu
                     GetSuccessTelemetry(queryCanPaste, decompileId, json, pasteType, bicep: null));
             }
 
-            // Show disclaimer and completion
-            string disclaimerMessage = "";
-            if (needsDisclaimer)
-            {
-                disclaimerMessage = $"{LangServerResources.DecompileAsPaste_AutoConvertWarning}\n{BicepDecompiler.DecompilerDisclaimerMessage}";
-                Log(output, disclaimerMessage);
-            }
-
             // Get Bicep output from the main file (all others are currently ignored)
             string bicepOutput = filesToSave.Single(kvp => BicepDummyUri.Equals(kvp.Key)).Value;
 
             // Ensure ends with newline
             bicepOutput = bicepOutput.TrimEnd() + "\n";
 
-            // Return result
+            // Show disclaimer and return result
+            Log(output, DisclaimerMessage);
             return new ResultAndTelemetry(
-                new BicepDecompileForPasteCommandResult(decompileId, output.ToString(), pasteType, null, bicepOutput, disclaimerMessage),
-                // Don't log telemetry if we're just determining if we can paste, because this will happen a lot
-                //   (on changing between editors for instance)
+                new BicepDecompileForPasteCommandResult(decompileId, output.ToString(), pasteType, null, bicepOutput, DisclaimerMessage),
                 GetSuccessTelemetry(queryCanPaste, decompileId, json, pasteType, bicepOutput));
         }
 
-        private static void Log(StringBuilder output, string message)
-        {
-            output.AppendLine(message);
-            Trace.TraceInformation(message);
-        }
-
         private BicepTelemetryEvent? GetSuccessTelemetry(bool queryCanPaste, string decompileId, string json, string? pasteType, string? bicep)
```

Comment:
Is it worth including including on how long the conversion took? Paste is probably very sensitive to delays, and the decompiler code hasn't had much thought put into optimization.

Reply:
I haven't ever seen it take very long, although I haven't tried stress testing with large files.  However, I've written code to show a notification in the event that it's taking too long but decided to break it out into separate code: https://github.com/Azure/bicep/issues/9597.  Will also do some stress testing here.

Note also that we're not currently actually blocking paste (we can't), we're running asynchronously after the paste.  If the editor has changed by the time the decompile finishes, I abort the conversion.

Reply:
Also, will consider adding time info to the event in #9597, but our vscode telemetry automatically adds time taken to all command events.